src/js/util/ReadLine.js

```diff
@@ -0,0 +1,23 @@
+import * as readline from "node:readline/promises";
+import { stdin as input, stdout as output } from "node:process";
+
+export const readLine = async (query) => {
+  const rl = readline.createInterface({ input, output });
+
+  const answer = await rl.question(query);
+
+  rl.close();
+
+  return answer;
+};
+
+export const readInteger = async (query) => {
+  const input = await readLine(query);
+  const inputInt = parseInt(input);
+
+  if (Number.isNaN(inputInt)) {
+    throw new Error("This input must be integer");
+  }
+
+  return inputInt;
+};
```

Comment:
정수 입력, 쉼표 파싱 등을 함수로 분리해 구현하셨는데 readLine에 입력값 검증과 입력값 가공 로직을 주입할 수 있도록 리팩터링하면 보다 확장성있는 입력 함수가 될 것 같아요 😄 
```suggestion
export const readLine = async (options: { 
  query,
  validate,
  transform,
}) => {
  const rl = readline.createInterface({ input, output });

  const answer = await rl.question(options.query);

  rl.close();

 // ...
};

Reply:
```javascript
export const readLine = async ({ query, validate, transform }) => {
  const rl = readline.createInterface({ input, output });

  const answer = await rl.question(query + "\n");

  rl.close();

  if (validate && !validate(answer)) {
    throw new Error();
  }

  if (transform) {
    return transform(answer);
  }

  return answer;
};
```

위와 같이 변경해보았습니다!